src/Library/demos/Web View/main.vala

```diff
@@ -0,0 +1,76 @@
+#! /usr/bin/env -S vala workbench.vala --pkg gtk4 --pkg libadwaita-1 --pkg webkitgtk-6.0
+
+public void main () {
+  var button_back = (Gtk.Button) workbench.builder.get_object ("button_back");
+  var button_forward = (Gtk.Button) workbench.builder.get_object ("button_forward");
+  var button_reload = (Gtk.Button) workbench.builder.get_object ("button_reload");
+  var button_stop = (Gtk.Button) workbench.builder.get_object ("button_stop");
+
+  var url_bar = (Gtk.Entry) workbench.builder.get_object ("url_bar");
+  var web_view = (WebKit.WebView) workbench.builder.get_object ("web_view");
+
+  web_view.bind_property ("uri", url_bar.buffer, "text", DEFAULT);
+  web_view.load_uri ("https://www.gnome.org/");
+
+  url_bar.activate.connect (() => {
+    string url = url_bar.buffer.text;
+    string? scheme = Uri.peek_scheme (url);
+    if (scheme == null) {
+      url = @"http://$url";
+    }
+    web_view.load_uri (url);
+  });
+
+  button_back.clicked.connect (() => web_view.go_back ());
+  button_forward.clicked.connect (() => web_view.go_forward ());
+  button_reload.clicked.connect (() => web_view.reload ());
+  button_stop.clicked.connect (() => web_view.stop_loading ());
+
+  web_view.load_changed.connect ((load_event) => {
+    switch (load_event) {
+      case STARTED:
+        message ("Loading page started");
+        break;
+      case FINISHED:
+        message ("Loading page finished");
+        break;
+      case COMMITTED:
```

Comment:
These could just be left out, as they don't do anything, right? Or are they for demonstrative purposes?

Reply:
I included them because otherwise the compiler would throw a warning saying that there are missing enum values in the switch statement. I preferred to add them to avoid confusion from the users.

Reply:
In that case add a `default` case.

Reply:
But to replace the FINISHED and COMMITTED cases :)

Reply:
Ohhhh that makes sense, sorry then haha :)